src/snapred/backend/data/Indexer.py

```diff
@@ -342,68 +303,111 @@ def addIndexEntry(self, entry: IndexEntry):
         Will save at the version on the index entry.
         If the version is invalid, will throw an error and refuse to save.
         """
-        if not self.isValidVersion(entry.version):
-            raise RuntimeError(f"Invalid version {entry.version} on index entry.  Save failed.")
-
+        entry.version = self._flattenVersion(entry.version)
         self.index[entry.version] = entry
         self.writeIndex()
 
     ## RECORD READ / WRITE METHODS ##
 
     def createRecord(self, *, version, **other_arguments):
         record = RECORD_TYPE[self.indexerType](
-            version=self.thisOrNextVersion(version),
+            version=self._flattenVersion(version),
             **other_arguments,
         )
         record.calculationParameters.version = record.version
         return record
 
-    def _determineRecordType(self, version: Optional[int] = None):
-        version = self.thisOrCurrentVersion(version)
+    def _determineRecordType(self, version: int):
         recordType = None
-        if version == VERSION_DEFAULT:
+        if version == self.defaultVersion():
             recordType = DEFAULT_RECORD_TYPE.get(self.indexerType, None)
         if recordType is None:
             recordType = RECORD_TYPE[self.indexerType]
         return recordType
 
-    def readRecord(self, version: Optional[int] = None) -> Record:
+    def readRecord(self, version: int) -> Record:
         """
         If no version given, defaults to current version
         """
-        version = self.thisOrCurrentVersion(version)
         filePath = self.recordPath(version)
         record = None
         if filePath.exists():
             record = parse_file_as(self._determineRecordType(version), filePath)
+        else:
+            raise FileNotFoundError(
+                f"No record found at {filePath} for version {version}, latest version is {self.currentVersion()}"
+            )
         return record
 
+    def _flattenVersion(self, version: Version):
+        flattenedVersion = None
+        if version == VersionState.DEFAULT:
+            flattenedVersion = self.defaultVersion()
+        elif version == VersionState.LATEST:
+            flattenedVersion = self.currentVersion()
+        elif version == VersionState.NEXT:
+            flattenedVersion = self.nextVersion()
+        elif isinstance(version, int):
+            flattenedVersion = version
+        else:
+            raise ValueError(f"Version must be an int or {VersionState.values()}, not {version}")
+
+        if flattenedVersion is None:
+            raise ValueError(
+                f"No available versions found during lookup using: "
+                f"v={version}, index={self.index}, dir={self.dirVersions}"
+            )
+        return flattenedVersion
+
+    def versionExists(self, version: Version):
+        return self._flattenVersion(version) in self.index
+
+    def writeNewRecord(self, record: Record, entry: IndexEntry):
```

Comment:
`writeIndexAndRecord`?  (and swap order of arguments)

Reply:
how about `writeNewVersion`?  These files need to be written in tandem due to current refresh behaviour.  
(semi-related: We should potentially just always read from disk)